src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs

```diff
@@ -54,8 +61,62 @@ public object Convert(object value, Type targetType, object parameter, CultureIn
 
             var iconBitmapImage = new BitmapImage();
             iconBitmapImage.BeginInit();
-            iconBitmapImage.UriSource = iconUrl;
 
+            BitmapSource imageResult;
+
+            if (values.Length > 1
+                && values[1] != null
+                && values[1] != System.Windows.DependencyProperty.UnsetValue
+                && iconUrl.IsFile
+                && !string.IsNullOrEmpty(iconUrl.Fragment)
+                && iconUrl.Fragment.Length > 1) // is it an embedded icon?
+            {
+                try
+                {
+                    var par = values[1] as PackageArchiveReader;
+                    var iconEntry = Uri.UnescapeDataString(iconUrl.Fragment).Substring(1);
```

Comment:
Why skipping the first char in substring? Comment may be useful.

Reply:
Because I want to skip the '#' in a URI fragment. Added a comment in this line